content/en/blog/_posts/2021-06-08-Kubeadm-Avoid-Default-Address.md

```diff
@@ -0,0 +1,285 @@
+kube-system   coredns-558bd4d5db-pgn9l        1/1     Running   0          71s   10.244.0.3       init1   <none>           <none>
+kube-system   coredns-558bd4d5db-xjdlp        1/1     Running   0          71s   10.244.0.2       init1   <none>           <none>
+kube-system   etcd-init1                      1/1     Running   0          83s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-apiserver-init1            1/1     Running   0          83s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-controller-manager-init1   1/1     Running   0          83s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-flannel-ds-hqtfh           1/1     Running   0          35s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-proxy-bn8s4                1/1     Running   0          71s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-scheduler-init1            1/1     Running   0          83s   192.168.56.101   init1   <none>           <none>
+```
+
+
+Next I took the `kubeadm join` command noted above and ran it on the
+worker node.  That reported success.  After waiting a little while I
+examined all the pods again and found some running on the worker node
+(for which I use the example name join1 and address 192.168.56.102
+here).
+
+```shell
+root@init1:~# kubectl get Pod -A -o wide
+NAMESPACE     NAME                            READY   STATUS    RESTARTS   AGE     IP               NODE    NOMINATED NODE   READINESS GATES
+kube-system   coredns-558bd4d5db-pgn9l        1/1     Running   0          2m38s   10.244.0.3       init1   <none>           <none>
+kube-system   coredns-558bd4d5db-xjdlp        1/1     Running   0          2m38s   10.244.0.2       init1   <none>           <none>
+kube-system   etcd-init1                      1/1     Running   0          2m50s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-apiserver-init1            1/1     Running   0          2m50s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-controller-manager-init1   1/1     Running   0          2m50s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-flannel-ds-hqtfh           1/1     Running   0          2m2s    192.168.56.101   init1   <none>           <none>
+kube-system   kube-flannel-ds-nrn6n           1/1     Running   0          15s     192.168.56.102   join1   <none>           <none>
+kube-system   kube-proxy-bn8s4                1/1     Running   0          2m38s   192.168.56.101   init1   <none>           <none>
+kube-system   kube-proxy-wxvpg                1/1     Running   0          15s     192.168.56.102   join1   <none>           <none>
+kube-system   kube-scheduler-init1            1/1     Running   0          2m50s   192.168.56.101   init1   <none>           <none>
+```
```

Comment:
How would you wrap this up?

Maybe a paragraph to say that you hope it's useful to show how kubeadm troubleshooting works, and also in case someone is facing a similar problem?

Reply:
Added a draft.